src/utransport.rs

```diff
@@ -152,37 +168,30 @@ impl UPClientZenoh {
                 Ok(sample) => {
                     // Get the encoding of UPayload
                     let Some(encoding) = UPClientZenoh::to_upayload_format(&sample.encoding) else {
-                        let msg = "Unable to get the encoding".to_string();
-                        log::error!("{msg}");
-                        block_on(
-                            resp_callback.on_error(UStatus::fail_with_code(UCode::INTERNAL, msg)),
-                        );
+                        invoke_block_callback(&resp_callback, Err("Unable to get the encoding"));
```

Comment:
Should this be blocking? 🤔

Could it be non-blocking? What if I want to send out N requests one after another in the uStreamer.

Reply:
In fact, this does nothing to your requests since this is inside the zenoh_callback, which means you can call several requests without blocking.

Reply:
Gotcha. That makes sense.